sympy/sets/handlers/intersection.py

```diff
@@ -377,7 +377,7 @@ def _solution_union(exprs, sym):
 
             range_set = S.EmptySet
 
-            if all(i.is_real for i in (new_sup, new_inf)):
+            if all(i.is_extended_real for i in (new_sup, new_inf)):
```

Comment:
This is effectively the same as the fix suggested at https://github.com/sympy/sympy/issues/24726#issuecomment-1476742742 (which is to remove the code in the below `else` block), because the bounds of `Interval` should always be extended real by definition (non-real intervals are not supported). It would be better to just delete this if condition completely and also the code in the `else` block below.